src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { forwardRef, useRef } from "react";
+import cx from "classnames";
+import { useMergeRefs } from "../../hooks";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import { getTestId } from "../../tests/test-ids-utils";
+import { ComponentDefaultTestId } from "../../tests/constants";
+import styles from "./Badge.module.scss";
+import { Direction } from "./BadgeConstants";
+import Indicator, { IndicatorProps } from "../Indicator/Indicator";
+import Counter, { CounterProps } from "../Counter/Counter";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { camelCase } from "lodash-es";
+
+type Position = {
+  vertical: Direction.TOP | Direction.BOTTOM;
+  horizontal: Direction.LEFT | Direction.RIGHT;
+};
+
+type BadgeOptions = typeof Indicator | typeof Counter;
+
+export interface BadgeProps extends VibeComponentProps, React.PropsWithChildren {
+  badgeComponent?: BadgeOptions;
```

Comment:
I think the component is redundant => can be only badge

Reply:
Also in the end we decided that the badge will be not a component it will be a type enum (counter or indication)